examples/azure/client_factory.md

```diff
@@ -0,0 +1,133 @@
+## Azure SDK Client Factory 
+
+This documentation provides and overview of the `client_factory.go` module, targeted use cases, and behaviors. 
+
+## Multi-cloud environment support
+
+In order to better support Azure customers, we needed to support both Public and sovereign cloud environments (includes USGovernment, Germany, China, and Azure Stack).  To do this, we need to use the correct base URI's for the Azure management plane when using the REST API (either directly or via SDK). For the Go SDK, this can be accomplished by using the `WithBaseURI` suffixed calls when creating service clients.
```

Comment:
NIT: We should write this from the voice of Terratest, the open source project, and not Microsoft, the company. 

```suggestion
The Azure APIs need to support both Public and sovereign cloud environments (includes USGovernment, Germany, China, and Azure Stack).  To do this, we need to use the correct base URI's for the Azure management plane when using the REST API (either directly or via SDK). For the Go SDK, this can be accomplished by using the `WithBaseURI` suffixed calls when creating service clients.
```

Reply:
Took a stab at making this in a more "active" voice...not sure how to describe without losing detail.